### ISSUE TITLE
Investigate Merge Blocking Plugin Enhancements To Better Enforce Release Freeze Deadlines

### ISSUE THREAD

**Issue by jeremyrickard** [kubernetes/sig-release#1389]
During the 1.20 Release, there were a merges to kubernetes/kubernetes that occurred post code freeze and post test freeze that had not been communicated to the release team. At each of these, people with appropriate permissions (lgtm, approve, milestone maintainer) can simply bypass the process and land things post freeze. These  have also occurred in past cycles, and lead to extra churn and toil for the Release Team, and in general, represent non-communicated risk to the release. 

The current merge blocking capabilities are too broad to be applied, we would really only want to apply them to the main branch during these freeze periods, until the release branch is cut. At that point, risk is descoped because cherry pick approval is required. Merge blocking is also currently pinned to an issue that when closed will drop the merge blockade. We very likely would need to make this team based (release team leads?) to make any blockade more effective.  We should also probably work in parallel to what the reliability working group is proposing for quality blocking, although this would be a different approach/purpose. 

/sig release
/milestone v1.21
/assign @jeremyrickard 
/cc @kubernetes/sig-release-leads
